File: src/main/scala/com/rawlabs/das/jira/DASJira.scala
/*
 * Copyright 2024 RAW Labs S.A.
 *
 * Use of this software is governed by the Business Source License
 * included in the file licenses/BSL.txt.
 *
 * As of the Change Date specified in that file, in accordance with
 * the Business Source License, use of this software will be governed
 * by the Apache License, Version 2.0, included in the file
 * licenses/APL.txt.
 */

package com.rawlabs.das.jira

import com.rawlabs.das.sdk.{DASFunction, DASSdk, DASTable}
import com.rawlabs.protocol.das.{FunctionDefinition, TableDefinition}
import com.typesafe.scalalogging.StrictLogging

class DASJira(options: Map[String, String]) extends DASSdk with StrictLogging {

  override def tableDefinitions: Seq[TableDefinition] = ???

  override def functionDefinitions: Seq[FunctionDefinition] = Seq.empty

  override def getTable(name: String): Option[DASTable] = ???

  override def getFunction(name: String): Option[DASFunction] = None
}

File: src/main/java/com/rawlabs/das/jira/DASJira.java
package com.rawlabs.das.jira;

import com.rawlabs.das.jira.initializer.DASJiraInitializer;
import com.rawlabs.das.jira.tables.*;
import com.rawlabs.das.sdk.java.DASFunction;
import com.rawlabs.das.sdk.java.DASSdk;
import com.rawlabs.das.sdk.java.DASTable;
import com.rawlabs.protocol.das.FunctionDefinition;
import com.rawlabs.protocol.das.TableDefinition;

import java.util.List;
import java.util.Map;

public class DASJira implements DASSdk {

  private final Map<String, String> options;
  private final DASJiraTableManager tableManager;

  protected DASJira(Map<String, String> options) {
    DASJiraInitializer.initialize(options);
    tableManager = new DASJiraTableManager(options);
    this.options = options;
  }

  @Override
  public List<TableDefinition> getTableDefinitions() {
    return tableManager.getTableDefinitions();
  }

  @Override
  public List<FunctionDefinition> getFunctionDefinitions() {
    return List.of();
  }

  @Override
  public DASTable getTable(String name) {
    return tableManager.getTable(name);
  }

  @Override
  public DASFunction getFunction(String name) {
    return null;
  }
}
